MusicNotationKit/MusicNotationKit/Clef.swift

```diff
@@ -9,43 +9,45 @@
 public struct Clef {
 
     public let tone: Tone?
-    public let lineNumber: Double // Could be changed to Float. Don't need that precision.
+    /**
+     Starts from 0 on the first line (from the bottom). Ledger lines below that are negative.
+     Each increase by 1 moves a half step. i.e. 1 is the first space on the staff.
+     */
+    public let lineNumber: Int
 
     /**
      You can create a custom clef by providing a tone and line number.
      
      - parameter tone: The tone that the clef represents
      - parameter lineNumber: The number representing the location on the staff. The `lineNumber`
-        starts at 0 for the first line on the staff. All spaces are the number of the preceding
-        line + 0.5. Upper ledger lines are negative and lower ledger lines are positive.
+        starts at 0 for the first line on the staff (from the bottom). Lower ledger lines are negative 
+        and upper ledger lines are positive. Each addition of 1 increases by a half step. This brings
+        you to either the next space or line.
      */
-    public init(tone: Tone, lineNumber: Double) throws {
-        try self.init(tone: .some(tone), lineNumber: lineNumber)
+    public init(tone: Tone, lineNumber: Int) {
+        self.init(tone: .some(tone), lineNumber: lineNumber)
     }
 
-    private init(tone: Tone?, lineNumber: Double) throws {
-        guard lineNumber.truncatingRemainder(dividingBy: 0.5) == 0 else {
-            throw ClefError.invalidLineNumber
-        }
+    private init(tone: Tone?, lineNumber: Int) {
         self.tone = tone
         self.lineNumber = lineNumber
     }
 
-    public static let treble = try! Clef(tone: Tone(noteLetter: .g, octave: .octave4), lineNumber: 3)
-    public static let bass = try! Clef(tone: Tone(noteLetter: .f, octave: .octave3), lineNumber: 1)
-    public static let tenor = try! Clef(tone: Tone(noteLetter: .c, octave: .octave4), lineNumber: 1)
-    public static let alto = try! Clef(tone: Tone(noteLetter: .c, octave: .octave4), lineNumber: 2)
+    public static let treble = Clef(tone: Tone(noteLetter: .g, octave: .octave4), lineNumber: 3)
+    public static let bass = Clef(tone: Tone(noteLetter: .f, octave: .octave3), lineNumber: 1)
+    public static let tenor = Clef(tone: Tone(noteLetter: .c, octave: .octave4), lineNumber: 1)
+    public static let alto = Clef(tone: Tone(noteLetter: .c, octave: .octave4), lineNumber: 2)
     /// Un-pitched (drums, percussion, etc.)
-    public static let neutral = try! Clef(tone: nil, lineNumber: 2)
+    public static let neutral = Clef(tone: nil, lineNumber: 2)
```

Comment:
Huh. Good point. It doesn't really have a function. I mean, I was seeing it just as where it will be drawn. Making it an optional seems like overkill, so I feel like it's fine to do this. What do you think?